web/app/js/components/util/ApiHelpers.jsx

```diff
@@ -118,10 +118,11 @@ const ApiHelpers = (pathPrefix, defaultMetricsWindow = '1m') => {
     metricsWindow = window;
   };
 
-  const urlsForResource = (type, namespace) => {
+  const urlsForResource = (type, namespace, includeTcp) => {
     // Traffic Performance Summary. This retrieves stats for the given resource.
     let baseUrl = '/api/tps-reports?resource_type=' + type;
-    return !namespace ? baseUrl + '&all_namespaces=true' : baseUrl + '&namespace=' + namespace;
+    let resourceUrl = !namespace ? baseUrl + '&all_namespaces=true' : baseUrl + '&namespace=' + namespace;
+    return !includeTcp ? resourceUrl : resourceUrl + '&tcp_stats=true';
```

Comment:
There's a lot happening in the ternary operator statements in lines 124-125 now that we have introduced includeTcp. It took a second for me to parse what was going on. You might consider breaking these out into `if/else` blocks for readability. Up to you.

Reply:
👍 

```js
let resourceUrl = baseUrl;
if namespace {
  resourceUrl += '&namespace=' + namespace;
} else {
  resourceUrl += '&all_namespaces=true';
}
if includeTcp {
  resourceUrl += &tcp_stats=true;
}

```